couchdb_cleaner.py

```diff
@@ -0,0 +1,176 @@
+         if(%s) {
+            emit(doc);
+         }
+     }
+''' % items
+
+
+def make_statement(field, value, sub_key=None, sub_query=False):
+    """Returns string representation of a javascript object
+       and his property, and checks if a property is equal to a value
+    :param field: a property of a object
+    :param value: value which will be checked
+    :param sub_key: a sub property of a property
+    :param sub_query: if True, return key, and a sub key of object
+    :return: string representation of a javascript object and his keys
+
+    Example without subquery
+    >>> make_statement('_id', '12345')
+    doc._id == '12345'
+
+    Example with subquery
+    >>> make_statement('dog', 'Marti', 'pets', sub_query=True)
+    doc.pets.dog == 'Marti'
+    """
+    if sub_query:
+        return "doc.{}.{} == '{}'".format(sub_key, field, value)
+    return "doc.{} == '{}'".format(field, value)
+
+
+def convert_to_dict(data):
+    """Convert a string representation of dictionary
+       into python dictionary type
+    :param data: An string representation of a dictionary
+    :return: dictionary object
+
+    >>> convert_to_dict("{'_id': '1234'}")
+    {'_id': '1234'}
+    """
+    return ast.literal_eval(data)
+
+
+def has_sub_query(query):
+    """Checks whether an keys, has a subquery
+    :param query: Checks whether an value of a key is a dictionary
+    :return: if value is a dictionary, returns True otherwise False
+     >>> my_dict = {'key': {'martin': True}}
+     >>> has_sub_query(my_dict)
+     True
+    """
+    if isinstance(query, dict):
+        return True
+    return False
+
+
+def make_query(parameters):
+    """Build a main query for CoachDB
+    :param parameters: A dictionary object
+    :return: An string representation of logical statement in javascript
+
+    >>> make_query({'doc_type': 'Lot', 'lotType' : {'procurement': 'ok', 'status': '200'}})
+    (doc.lotType.status == '200' && doc.lotType.procurement == 'ok' && doc.doc_type == 'Lot')
+    """
+    sub_query = []
+    for key in parameters.keys():
+        if has_sub_query(parameters[key]):
+            for k, v in parameters[key].items():
+                sub_query.append(make_statement(k, v, key, True))
+            parameters.pop(key)
+
+    prepare_query = [make_statement(k, v) for k, v in parameters.items()]
+    output = sub_query + prepare_query
+
+    if len(output) == 1:
+        return base_view(''.join(output))
+    return base_view(' && '.join(output))
+
+
+def get_metadata(obj):
+    """Returns object's ID and REV keys
+    :param obj: A CoachDB object
+    :return: a dictionary with object's information
+    """
+    return {'_id': obj['id'], '_rev': obj['key']['_rev']}
+
+
+def welcome_message():
+    """User friendly message"""
+    print '=' * 35
+    print 'Welcome in Database Cleaner\nPlease enter an credentials under'
+    print '=' * 35
+
+
+def make_database_connection(database_url):
+    """Connects to a CoachDB database
```

Comment:
CouchDB

Reply:
@oleksiyVeretiuk Yeah, sorry it's the my mistake, I'll fix it in 20-30 minutes. Thank you